src/Nowruz/pages/sign-up/Onboarding/index.tsx

```diff
@@ -1,115 +1,119 @@
-        <img className={css.headerImage} src={isMobile ? '/icons/logo.svg' : '/icons/logo-text.svg'} />
-        <IconDropDown iconItems={items} type="users" accounts={accounts} />
-      </div>
-      <div className="flex flex-col items-center pb-4 sb:h-screen">
-        <div className={css.container}>
-          <Steper
-            components={[
-              { Component: <Welcome />, skippable: false },
-              { Component: <Causes />, skippable: false },
-              { Component: <Skills />, skippable: false },
-              { Component: <City />, skippable: false },
-              { Component: <ImageBio />, skippable: false },
-            ]}
-          />
-        </div>
-      </div>
-    </UserProvider>
-  );
-};
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { CurrentIdentity, identities, logout, OrgMeta, UserMeta } from 'src/core/api';
+import { nonPermanentStorage } from 'src/core/storage/non-permanent';
+import { Causes } from 'src/Nowruz/modules/Auth/containers/onboarding/Causes';
+import { City } from 'src/Nowruz/modules/Auth/containers/onboarding/City';
+import { CreateOrganization } from 'src/Nowruz/modules/Auth/containers/onboarding/CreateOrganization';
+import { ImageBio } from 'src/Nowruz/modules/Auth/containers/onboarding/ImageBio';
+import { OpWelcome } from 'src/Nowruz/modules/Auth/containers/onboarding/OpWelcome';
+import { OrganizationCauses } from 'src/Nowruz/modules/Auth/containers/onboarding/OrganizationCauses';
+import { OrganizationContact } from 'src/Nowruz/modules/Auth/containers/onboarding/OrganizationContact';
+import { OrganizationLogo } from 'src/Nowruz/modules/Auth/containers/onboarding/OrganizationLogo';
+import { OrganizationType } from 'src/Nowruz/modules/Auth/containers/onboarding/OrganizationType';
+import { Skills } from 'src/Nowruz/modules/Auth/containers/onboarding/Skills';
+import Steper from 'src/Nowruz/modules/Auth/containers/onboarding/Stepper';
+import { Welcome } from 'src/Nowruz/modules/Auth/containers/onboarding/Welcome';
+import { UserProvider, useUser } from 'src/Nowruz/modules/Auth/contexts/onboarding/sign-up-user-onboarding.context';
+import { AccountItem } from 'src/Nowruz/modules/general/components/avatarDropDown/avatarDropDown.types';
+import { IconDropDown } from 'src/Nowruz/modules/general/components/iconDropDown';
+import store, { RootState } from 'src/store';
+import { removeIdentityList, setIdentityList } from 'src/store/reducers/identity.reducer';
+import { setIdentity } from 'src/store/reducers/profile.reducer';
+
+import css from './onboarding.module.scss';
+
+export const Onboarding = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const primary = useSelector<RootState, CurrentIdentity | undefined>(state => {
+    return state.identity.entities.find(identity => identity.current);
+  });
+
+  useEffect(() => {
```

Comment:
could you please rewrite this piece of code with `async/await` (I know it's not a new change, but it's better to change everywhere we see `.then`)
```useEffect(() => {
  const fetchData = async () => {
    const resp = await identities();
    await dispatch(setIdentityList(resp));
    
    const user = {
      id: primary?.id,
      username: (primary?.meta as UserMeta).username || (primary?.meta as OrgMeta).shortname || '',
      email: (primary?.meta as UserMeta).email || (primary?.meta as OrgMeta).email,
    };
    await dispatch(setIdentity(user));
  };

  fetchData();
}, []);
```